app/elements/pages/page-browse.html

```diff
@@ -238,7 +238,15 @@ <h2 class="flex"><span>[[pageTitle]]</span> <span class="count">[[_elementCount]
       updateSearch: function() {
         var params = this._parseQueryString();
         if (params && params.q) {
-          this.async(function() { this.$.query.focus(); });
+          this.async(function() {
+            this.$.query.focus();
+
+            // make sure to force the cursor to the end of the
+            // input. Only an issue in Firefox.
+            if (this.$.query.setSelectionRange) {
```

Comment:
Yeah I think `setSelectionRange` (basic) is well supported in all the browsers that Polymer supports.  We can probably remove the check here.